nix/package.nix

```diff
@@ -0,0 +1,25 @@
+# Override ladybird from nixpkgs with the source directory as the source
+# and latest revision (or "dirty") as the version. Unless new dependencies
+# are introduced, this should work just fine. If a new dependency is added
+# and builds are broken, simply adding the missing dependencies in buildInputs
+# inside of the attribute should do the trick.
+{
+  self,
+  lib,
+  ladybird,
+  ...
+}:
+ladybird.overrideAttrs {
+  # Reproducible source path
+  src = builtins.path {
```

Comment:
consider `lib.fileset` as well

Reply:
I feel like it's not worth the effort trying to filter things while I'm not really familiar  with the structure of the project. Though it *would* make sense to apply  a filter here to at least omit the known useless stuff.